src/dune_manager/dune_manager.ml

```diff
@@ -455,8 +480,21 @@ module Client = struct
       Result.try_with (fun () -> Unix.connect fd (Unix.ADDR_INET (addr, port)))
     in
     let socket = Unix.out_channel_of_descr fd in
+    let input = Stream.of_channel (Unix.in_channel_of_descr fd) in
+    let rec thread input =
+      match
+        let+ command = read input in
+        Log.infof "dune-cache command: %a" pp_command command;
+        handle command
+      with
+      | Result.Error e -> Log.infof "dune-cache read error: %s" e
+      | Result.Ok () -> (thread [@tailcall]) input
```

Comment:
This attribute is not necessary here. The compiler is clever enough and this is not a critical function either. It's better to keep the code simple.

Reply:
It's more of a security towards the programmer : since it's an infinite recursion, I want to be warned if I mess up and make it non-terminal somehow.

Reply:
I  see. That makes sense, but at the same time I'd like that we are consistent across the code base. Currently there are other such functions that don't have the attribute. So we should either add it everywhere or have it nowhere.

Personally, I'd prefer to not have it as I feel like in general I would be careful about keeping such function tail-recursive and so the attribute just adds noise to the code. But I'm not opposed to it either if the general consensus is that it is valuable. Perhaps we could discuss this on dune-dev to see what others think?